components/button/test/button.html

```diff
@@ -0,0 +1,143 @@
+<!doctype html>
+<html lang="en">
+	<head>
+		<meta charset="utf-8">
+		<meta name="viewport" content="width=device-width, minimum-scale=1, initial-scale=1, user-scalable=yes">
+		<title>d2l-button unit tests</title>
+		<script src="/node_modules/@webcomponents/webcomponentsjs/webcomponents-loader.js"></script>
+		<script src="/node_modules/mocha/mocha.js"></script>
+		<script src="/node_modules/chai/chai.js"></script>
+		<script src="/node_modules/@polymer/test-fixture/test-fixture.js"></script>
+		<script src="/node_modules/wct-mocha/wct-mocha.js"></script>
+		<script type="module" src="../button.js"></script>
+	</head>
+	<body>
+
+		<test-fixture id="basic">
+			<template>
+				<d2l-button>Normal Button</d2l-button>
+			</template>
+		</test-fixture>
+
+		<test-fixture id="primary">
+			<template>
+				<d2l-button primary="">Primary Button</d2l-button>
+			</template>
+		</test-fixture>
+
+		<test-fixture id="disabled">
+			<template>
+				<d2l-button disabled="">Disabled Button</d2l-button>
+			</template>
+		</test-fixture>
+
+		<test-fixture id="primaryDisabled">
+			<template>
+				<d2l-button disabled="" primary="">Disabled Primary Button</d2l-button>
+			</template>
+		</test-fixture>
+
+	<script type="module">
+			import { runAxe } from '../../../tools/a11y-test-helper.js';
+
+			describe('d2l-button', () => {
+				let button;
+
+				describe('basic', () => {
+
+					beforeEach(async() => {
+						button = fixture('basic');
+						await button.updateComplete;
+					});
+
+					it('should execute onclick when clicked', async() => {
+						return new Promise(resolve => {
+							button.addEventListener('click', resolve, {once: true});
+							button.click();
+						});
+					});
+
+					it('should pass all axe tests', async() => {
+						await runAxe(button);
+					});
+
+					it('should pass all axe tests when focused', async() => {
+						return new Promise((resolve, reject) => {
+							button.addEventListener('focus', async() => {
+								try {
+									await runAxe(button);
+									resolve();
+								} catch (e) {
+									reject(e);
+								}
+							});
+							button.shadowRoot.querySelector('button').focus();
+						});
+					});
+
+				});
+
+				describe('primary', () => {
+
+					beforeEach(async() => {
+						button = fixture('primary');
+						await button.updateComplete;
+					});
+
+					it('should execute onclick when clicked', async() => {
+						return new Promise(resolve => {
+							button.addEventListener('click', resolve, {once: true});
+							button.click();
+						});
+					});
```

Comment:
I don't really see a need to duplicate this test again for primary buttons -- they're simply a style thing.